File: examples/adopt-go/greet/greet.go
// Package greet is the foreign Go package adopted without upstream changes:
// stdlib-only sources plus platform-selected files. Generation owns the
// Bazel package-level library; the Go toolchain selects per platform.
package greet

import "fmt"

// Hello returns a greeting for name.
func Hello(name string) string {
	return fmt.Sprintf("hello %s", name)
}

File: examples/adopt-cpp/greet/greet.cc
// Foreign C++ greeting implementation: adopted without upstream changes.
#include "examples/adopt-cpp/greet/greet.h"

#include <string>

#include "examples/adopt-cpp/greet/helper.h"

std::string Greet(const std::string& name) {
  return "hello " + name + HelperSuffix();
}
